### PR TITLE
Get the Leap Motion working again

### PR DESCRIPTION
**Pull request by ctrlaltdavid**
This PR makes it so that the Leap can be used again.  The .js scripts need updating due to recent animation API changes; those will come in another PR.

### CODE REVIEW

**Comment by ctrlaltdavid** [comment]
https://worklist.net/20740

**Comment by hifi-gustavo** [comment]
Build console output for commit: 46cbc349d15d7f103e4568abb42fa183f2e9f7d8 available at: https://gist.github.com/508670bed545e4e53cdf<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6487.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6487.dmg

File: interface/src/scripting/ControllerScriptingInterface.cpp [hunk 1]
```diff
@@ -84,13 +84,12 @@ glm::vec2 ControllerScriptingInterface::getViewportDimensions() const {
     return qApp->getUiSize();
 }
 
-controller::InputController::Pointer ControllerScriptingInterface::createInputController(const QString& deviceName, const QString& tracker) {
-    // This is where we retreive the Device Tracker category and then the sub tracker within it
+controller::InputController* ControllerScriptingInterface::createInputController(const QString& deviceName, const QString& tracker) {
```
Comment (ZappoMan) [inline interface/src/scripting/ControllerScriptingInterface.cpp]: what's this change for?

**Comment by ZappoMan** [comment]
why are you changing the smart pointers into naked pointers?

**Comment by hifi-gustavo** [comment]
Build console output for commit: 6e076985f2ececbb2013cb14d7f72e346ce91394 available at: https://gist.github.com/508670bed545e4e53cdf<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6487.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6487.dmg

**Comment by ctrlaltdavid** [comment]
Smart pointers are still used internally in the C++ (stored in _inputControllers, and used in createInputController() and releaseInputController()); the naked pointers are only passed to JavaScript for its reference and communication back to C++. It was changed because createController() wasn't working.

**Comment by imgntn** [comment]
@ctrlaltdavid hey, i've been testing this on windows 10 with a leap that we have in the office.  the two scripts (laserpointer and leaphands) seem to work, although intermittently.  i'm not sure what the previous baseline functionality was that we're trying to get back to.  if it's simply working at all, then we're good to go (it doesn't work in release in my testing but works in this PR).  as far as being useful/robust, i don't think that these controls schemes are there yet, but i'm testing in a pretty bright office.  any tips as to best ways to test?  my arms often end up in really weird configurations  / loops when i use leaphands.  additionally, you have to uncheck the "Draw Anim Graph" option, which i'm not sure is something we can do programatically...?  thanks!

**Comment by imgntn** [comment]
@ctrlaltdavid - I spoke with @cozza13 about where functionality should be and working at all seems to be a major improvement.  the control script will need a lot of work to be usable (so did the hydra one).  I think we should merge this as long as the C++ stuff @ZappoMan was asking about is good to go.  Then anyone with a leap can start working on the control script.

**Comment by cozza13** [comment]
@ctrlaltdavid @murillodigital which Leap SDK are we using on the build machine?

**Comment by murillodigital** [comment]
@cozza13 looks like (2.2.1+24116) but since this has been not maintained for some time I'd have to double check all servers to confirm consistency.

However, please make it so leap is handled as a cmake external and assume no library is present otherwise.

**Comment by ctrlaltdavid** [comment]
@murillodigital Interface installer currently includes Leap.dll version 2.2.1, presumably with the SDK handled per \HiFi\interface\external\leapmotion\readme.txt (i.e., uses particular version of SDK installed on the build machine).
"handled as a cmake external" means move Leap SDK handling into \HiFi\cmake\externals\, i.e., automatically download latest SDK from Leap Motion when building? If so, merge this PR now and change to using a cmake external as a separate PR?

**Comment by imgntn** [comment]
@ctrlaltdavid i think that's a good plan.  let's merge this and then we'll fund a worklist item to do the external https://worklist.net/20745

**Comment by ctrlaltdavid** [comment]
Changed to "do not merge" because since the PR was submitted, MyAvatar.setJointModelPositionAndOrientation() has been removed from the JavaScript API, meaning that leapHands.js no longer works.

**Comment by imgntn** [comment]
@ctrlaltdavid -- are the C++ changes required for the leap to work at all? if so, i think we could isolate those, merge, and then do another worklist item for making the leaphands script better -- i think we are more interested in moving toward using it to grab / equip items, rather than full finger-level control

**Comment by ctrlaltdavid** [comment]
@imgntn Yes, the C++ changes are required to get the Leap Motion working at all: fixes a Leap Motion InputController not being able to be created and fixes the resulting controller not being updated.
Could certainly merge this PR as is and do a subsequent PR to update leapHands.js to do something else instead of the setJointModelPositionAndOrientation() calls.

**Pull request merged**